ipc_receivefile.c

```diff
@@ -418,16 +426,81 @@ int ipc_receive_pipe(char* fileName){
 	return 0;
 }
 
+int ipc_receive_queue(char* fileName){
+	int status;
+
+	mqd_t msg_queue;
+	struct mq_attr attrs;
+	int prio;
+	int receive_size;
+
+	struct timespec abs_timeout;
+	fileInit(fileName);
+	char * buffer = malloc (MAX_MSG_QUEUE_SIZE);
+	memset(&attrs, 0, sizeof attrs);
 
+	printf("Connecting to the queue\n");
 
+	do {
+		errno = 0;
+		msg_queue = mq_open( "queue", O_RDONLY, &attrs );
+		if (errno == ENOENT){
+			sleep (2);
+		}
+	}while (errno == ENOENT);
+
+	if (msg_queue == -1){
+		perror ("mq_open");
+		exit(EXIT_FAILURE);
+	}
+
+	clock_gettime(CLOCK_REALTIME, &abs_timeout);
+	abs_timeout.tv_sec += 1;
+
+	do {
+
+		receive_size = mq_timedreceive (msg_queue, buffer, MAX_MSG_QUEUE_SIZE, &prio,&abs_timeout);
+
+		if (receive_size == -1) {
+			if (errno == ETIMEDOUT) {
+				printf ("mq_timedreceive() timed out file fully transfered.\n");
+			 } else {
+				perror ("mq_timedreceive()");
```

Comment:
free(buffer) 

You should check every exit/return and also in ipc_sendfile

Reply:
Done!